mongomock/aggregate.py

```diff
@@ -509,6 +509,89 @@ def _handle_lookup_stage(in_collection, database, options):
     return in_collection
 
 
+def _handle_graph_lookup_stage(in_collection, database, options):
+    if not isinstance(options.get('maxDepth', 0), six.integer_types):
+        raise OperationFailure(
+            "Argument 'maxDepth' to $graphLookup must be a number")
+    if not isinstance(options.get('restrictSearchWithMatch', {}), dict):
+        raise OperationFailure(
+            "Argument 'restrictSearchWithMatch' to $graphLookup must be a Dictionary")
+    if not isinstance(options.get('depthField', ''), six.string_types):
+        raise OperationFailure(
+            "Argument 'depthField' to $graphlookup must be a string")
+    for operator in ('from', 'startWith', 'connectFromField', 'connectToField', 'as'):
+        if operator not in options:
+            raise OperationFailure(
+                "Must specify '%s' field for a $graphLookup" % operator)
+        if not isinstance(options[operator], six.string_types):
+            raise OperationFailure(
+                "Argument '%s' to $graphLookup must be string" % operator)
+        if operator in ('as', 'connectFromField', 'connectToField') and \
+                options[operator].startswith('$'):
+            raise OperationFailure(
+                "FieldPath field names may not start with '$'")
+        if operator == 'startWith' and \
+                not options[operator].startswith('$'):
+            raise OperationFailure(
+                "Field ref names must start with '$'")
+        if operator in ('connectFromField', 'startWith', 'as') and \
+                '.' in options[operator]:
+            raise NotImplementedError(
+                "Although '.' is valid in the '%s' "
+                'parameter for the $graphLookup stage of the aggregation '
+                'pipeline, it is currently not implemented in Mongomock.' % operator)
+
+    foreign_name = options['from']
+    start_with = options['startWith'][1:]
+    connect_from_field = options['connectFromField']
+    connect_to_field = options['connectToField']
+    local_name = options['as']
+    max_depth = options.get('maxDepth', None)
+    depth_field = options.get('depthField', None)
+    restrict_search_with_match = options.get('restrictSearchWithMatch', {})
+    foreign_collection = database.get_collection(foreign_name)
+    for doc in in_collection:
+        found_items = set()
+        if start_with in doc:
```

Comment:
Actually start_with is an expression, so it could be `{'$concat`: ['$field1', '$field2']}`, please use parse_expression. That way you can also accept dots in startWith.

Reply:
Done that after your previous comment on startWith being array and such :)